dasar/html/form.md

```diff
@@ -1,49 +1,104 @@
 
-## 2. Macam - Macam Tag Untuk Membuat Form
+## 2. Penjelasan Form
+Form digunakan untuk menerima data dari inputan user, biasanya diterapkan pada form login, pendaftaran, transaksi, dan sebagainya. 
 
-### 2.1. Form
+Form dalam web bisa disamakan dengan formulir di dunia nyata. Form dapat diisi, kemudian diproses dengan program tertentu.
 
-Untuk mulai membuat form, kita perlu membungkusnya menggunakan tag `form`, nantinya semua inputan akan kita masukkan ke dalam tag ini
+## 3. Cara Membuat Form
+
+Form di HTML dapat kita buat dengan tag `<form>`.
+nantinya semua inputan akan dimasukkan ke dalam tag ini
 
 ```html
-<form>
+<form >
   <!-- input form ada disini -->
 </form>
 ```
+Kode html diatas tidak akan menghasilkan apa-apa. 
+Karena kita belum membuat field-nya.
+
+Apa itu field ?
+`Field` adalah ruas yang dapat diisi dengan data.
+
+Contoh field :
+```html
+<form>
+  <input type="text" name="info" />
+</form>
+```
+Field memiliki beberapa attribut yang harus diberikan :
+-> `type` merupakan type dari field.
+-> `name` merupakan nama dari field yang akan menjadi kunci dan variabel di dalam program.
 
-### 2.2 Input
+### 4. Membuat Form Login
 
-Untuk menerima inputan dari user, kita bisa menggunakan tag `input` untuk menampilkan input text, checkbox, radio, button, dan lain lain
+Pada form login, terdapat beberapa field dan elemen : 
+-> Field untuk input username atau email
+-> Field untuk input password
+-> Checkbox untuk remember me
+-> Tombol untuk login
+
+### 4.1 Input
+
+Untuk menerima inputan dari user, kita bisa menggunakan tag `input`.
 
 ```html
 <form>
   <input type="text" />
+  
+  <input type="password" />
 
   <input type="checkbox" />
 
-  <input type="radio" />
-
-  <input type="button" />
+  <input type="submit" />
 </form>
 ```
-
-### 2.3 Label
-
-Label digunakan untuk memberikan label pada inputan yang kita buat, sehingga user tau data apa yang perlu dimasukkan
+### 4.2 Label
+Label digunakan untuk memberikan label pada inputan yang kita buat, sehingga user tau data apa yang perlu dimasukkan ke dalam form
 
 ```html
 <form>
-  <label for="username">username</label>
-  <input type="text" id="username" />
+  <label>Username:</label>
+  <input type="text" name="username" placeholder="username..." />
+
+  <label>Password:</label>
+  <input type="password" name="password" placeholder="password..." />
 
-  <label for="password">password</label>
-  <input type="password" id="password" />
+  <label>Remember me </label>
+  <input type="checkbox" name="remember" value="remember" />
 </form>
 ```
-
-disini kita perlu memberikan `id` pada `input` yang kita buat, kemudian kita tinggal menambahkan attribut `for` pada label yang mengacu pada id input yang telah dibuat, sehingga apabila label tersebut kita klik maka cursor akan langsung fokus ke input yang terkait
+### 4.3 Baris Kode 
```

Comment:
langkah yang ini maksudnya mau nunjukin keseluruhan code nya kan ? kalau gitu diganti aja judulnya biar lebih to the point

```suggestion
### 4.3 Hasil Akhir
```